stm32cubemx: Can't initiate JxBrowser engine

### Describe the bug
When clicking "GENERATE CODE" nothing happens, and I get  the error from title in the console log.
I'm not familiar with java but I guess newer cubemx versions require some browser like functionality which failed to init within java runtime due missing .so dependencies.

Startup Info:

```
log4j user configuration file not found: /home/akill/.stm32cubemx/log4j2.xml
Configure log4j with default settings from jar:file:/nix/store/3mv4p1lz0xmmy5cb3zz9d74vsmqyj58m-stm32cubemx-6.9.2/opt/STM32CubeMX/STM32CubeMX!/log4j/log4j2.stm32cubemx.xml
2023-12-05 21:06:19,347 [INFO] MicroXplorer:98 - [MX] MX Start == 60807442279392
2023-12-05 21:06:19,348 [INFO] MicroXplorer:653 - Detected Java Version = 17.0.7
2023-12-05 21:06:19,620 [INFO] ApplicationProperties:184 - Using Application install path: /nix/store/3mv4p1lz0xmmy5cb3zz9d74vsmqyj58m-stm32cubemx-6.9.2/opt/STM32CubeMX
2023-12-05 21:06:19,632 [INFO] DbMcusXml:78 - Set database path to: /nix/store/3mv4p1lz0xmmy5cb3zz9d74vsmqyj58m-stm32cubemx-6.9.2/opt/STM32CubeMX//db//mcu/
2023-12-05 21:06:19,632 [INFO] ApiDb:274 - Set plugin database path to: /nix/store/3mv4p1lz0xmmy5cb3zz9d74vsmqyj58m-stm32cubemx-6.9.2/opt/STM32CubeMX//db//plugins/boardmanager/
``` 

Relevant error log:
```
2023-12-05 21:06:23,473 [ERROR] JxBrowserEngine:69 - Can't initiate JxBrowser engine com.teamdev.jxbrowser.engine.MissingDependencyException: Missing dependencies have been detected:
	libGLESv2.so => libXext.so.6 libxcb.so.1 libX11.so.6 
	chromium => libXfixes.so.3 libasound.so.2 libXcomposite.so.1 libcairo.so.2 libxcb.so.1 libpango-1.0.so.0 libgio-2.0.so.0 libatspi.so.0 libatk-bridge-2.0.so.0 libXext.so.6 libdrm.so.2 libnspr4.so libxkbcommon.so.0 libnssutil3.so libcups.so.2 libgbm.so.1 libglib-2.0.so.0 libsmime3.so libatk-1.0.so.0 libgobject-2.0.so.0 libexpat.so.1 libdbus-1.so.3 libXdamage.so.1 libXrandr.so.2 libnss3.so libX11.so.6 
	libtoolkit.so => libgtk-3.so.0 libglib-2.0.so.0 libgobject-2.0.so.0 libX11.so.6 libgdk-3.so.0 
Exception in thread "Thread-3" java.lang.NullPointerException: Cannot invoke "javax.swing.JComponent.setVisible(boolean)" because "this.webAppContainer" is null
	at com.st.microxplorer.plugins.userauth.UserAuth.activatePlugin(UserAuth.java:200)
	at com.st.microxplorer.maingui.MainFrame.lambda$initialization$0(MainFrame.java:189)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

```2023-12-05 21:06:36,871 [ERROR] ProjectManagerView:731 - 
java.lang.NullPointerException: Cannot invoke "com.teamdev.jxbrowser.browser.Browser.mainFrame()" because "this.browser" is null
	at com.st.microxplorer.plugins.userauth.WebApp.isAuthenticated(WebApp.java:615) ~[?:?]
	at com.st.microxplorer.plugins.userauth.UserAuth.handleAuthentication(UserAuth.java:452) ~[?:?]
	at com.st.microxplorer.plugins.userauth.UserAuth.getSpecificTask(UserAuth.java:259) ~[?:?]
	at com.st.microxplorer.plugins.projectmanager.gui.ProjectChoiceTab.save(ProjectChoiceTab.java:1349) ~[?:?]
	at com.st.microxplorer.plugins.projectmanager.ProjectManagerView.generateCode(ProjectManagerView.java:1068) ~[?:?]
	at com.st.microxplorer.plugins.projectmanager.ProjectManagerView.getSpecificTask(ProjectManagerView.java:729) ~[?:?]
	at com.st.microxplorer.maingui.RibbonBar$1.actionPerformed(RibbonBar.java:363) ~[STM32CubeMX:?]
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) ~[?:?]
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) ~[?:?]
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) ~[?:?]
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) ~[?:?]
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279) ~[?:?]
	at java.awt.Component.processMouseEvent(Component.java:6626) ~[?:?]
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3389) ~[?:?]
	at java.awt.Component.processEvent(Component.java:6391) ~[?:?]
	at java.awt.Container.processEvent(Container.java:2266) ~[?:?]
	at java.awt.Component.dispatchEventImpl(Component.java:5001) ~[?:?]
	at java.awt.Container.dispatchEventImpl(Container.java:2324) ~[?:?]
	at java.awt.Component.dispatchEvent(Component.java:4833) ~[?:?]
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) ~[?:?]
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) ~[?:?]
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) ~[?:?]
	at java.awt.Container.dispatchEventImpl(Container.java:2310) ~[?:?]
	at java.awt.Window.dispatchEventImpl(Window.java:2780) ~[?:?]
	at java.awt.Component.dispatchEvent(Component.java:4833) ~[?:?]
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:722) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:716) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:399) ~[?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) ~[?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) ~[?:?]
	at java.awt.EventQueue$5.run(EventQueue.java:746) ~[?:?]
	at java.awt.EventQueue$5.run(EventQueue.java:744) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:399) ~[?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) ~[?:?]
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:743) ~[?:?]
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) ~[?:?]
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) ~[?:?]
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) ~[?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) ~[?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) ~[?:?]
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:90) ~[?:?]
```

### Steps To Reproduce

- Spawn shell with stm32cubemx using  `nix shell nixpkgs#stm32cubemx --impure`.
- Create CubeMX project
- Generate Code

### Expected behavior
Generated project folder with confirmation window.

### Notify maintainers
@angaz @wucke13 

### Metadata

```
❯ nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.2, NixOS, 24.05 (Uakari), 24.05.20231127.8cfef69`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - channels(akill): `"home-manager, nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/embedded/stm32/stm32cubemx/default.nix
+++ b/pkgs/development/embedded/stm32/stm32cubemx/default.nix
@@ -79,7 +79,7 @@ let
       sourceProvenance = with sourceTypes; [ binaryBytecode ];
       license = licenses.unfree;
       maintainers = with maintainers; [ angaz wucke13 ];
-      platforms = platforms.all;
+      platforms = [ "x86_64-linux" ];
     };
   };
   in

--- a/pkgs/development/embedded/stm32/stm32cubemx/default.nix
+++ b/pkgs/development/embedded/stm32/stm32cubemx/default.nix
@@ -1,84 +1,115 @@
 { fdupes
+, buildFHSEnv
 , fetchzip
 , icoutils
 , imagemagick
 , jdk17
 , lib
 , makeDesktopItem
-, stdenv
+, stdenvNoCC
 }:
 
 let
   iconame = "STM32CubeMX";
-in
-stdenv.mkDerivation rec {
-  pname = "stm32cubemx";
-  version = "6.10.0";
+  package = stdenvNoCC.mkDerivation rec {
+    pname = "stm32cubemx";
+    version = "6.10.0";
 
-  src = fetchzip {
-    url = "https://sw-center.st.com/packs/resource/library/stm32cube_mx_v${builtins.replaceStrings ["."] [""] version}-lin.zip";
-    sha256 = "sha256-B5Sf+zM7h9BiFqDYrLS0JdqZi3dGy6H9gAaJIN3izeM=";
-    stripRoot = false;
-  };
+    src = fetchzip {
+      url = "https://sw-center.st.com/packs/resource/library/stm32cube_mx_v${builtins.replaceStrings ["."] [""] version}-lin.zip";
+      sha256 = "sha256-B5Sf+zM7h9BiFqDYrLS0JdqZi3dGy6H9gAaJIN3izeM=";
+      stripRoot = false;
+    };
 
-  nativeBuildInputs = [ fdupes icoutils imagemagick ];
-  desktopItem = makeDesktopItem {
-    name = "STM32CubeMX";
-    exec = "stm32cubemx";
-    desktopName = "STM32CubeMX";
-    categories = [ "Development" ];
-    icon = "stm32cubemx";
-    comment = meta.description;
-    terminal = false;
-    startupNotify = false;
-    mimeTypes = [
-      "x-scheme-handler/sgnl"
-      "x-scheme-handler/signalcaptcha"
-    ];
-  };
+    nativeBuildInputs = [ fdupes icoutils imagemagick ];
+    desktopItem = makeDesktopItem {
+      name = "STM32CubeMX";
+      exec = "stm32cubemx";
+      desktopName = "STM32CubeMX";
+      categories = [ "Development" ];
+      icon = "stm32cubemx";
+      comment = meta.description;
+      terminal = false;
+      startupNotify = false;
+      mimeTypes = [
+        "x-scheme-handler/sgnl"
+        "x-scheme-handler/signalcaptcha"
+      ];
+    };
 
-  buildCommand = ''
-    mkdir -p $out/{bin,opt/STM32CubeMX,share/applications}
+    buildCommand = ''
+      mkdir -p $out/{bin,opt/STM32CubeMX,share/applications}
 
-    cp -r $src/MX/. $out/opt/STM32CubeMX/
-    chmod +rx $out/opt/STM32CubeMX/STM32CubeMX
+      cp -r $src/MX/. $out/opt/STM32CubeMX/
+      chmod +rx $out/opt/STM32CubeMX/STM32CubeMX
 
-    cat << EOF > $out/bin/${pname}
-    #!${stdenv.shell}
-    ${jdk17}/bin/java -jar $out/opt/STM32CubeMX/STM32CubeMX
-    EOF
-    chmod +x $out/bin/${pname}
+      cat << EOF > $out/bin/${pname}
+      #!${stdenvNoCC.shell}
+      ${jdk17}/bin/java -jar $out/opt/STM32CubeMX/STM32CubeMX
+      EOF
+      chmod +x $out/bin/${pname}
 
-    icotool --extract $out/opt/STM32CubeMX/help/${iconame}.ico
-    fdupes -dN . > /dev/null
-    ls
-    for size in 16 24 32 48 64 128 256; do
-      mkdir -pv $out/share/icons/hicolor/"$size"x"$size"/apps
-      if [ $size -eq 256 ]; then
-        mv ${iconame}_*_"$size"x"$size"x32.png \
-          $out/share/icons/hicolor/"$size"x"$size"/apps/${pname}.png
-      else
-        convert -resize "$size"x"$size" ${iconame}_*_256x256x32.png \
-          $out/share/icons/hicolor/"$size"x"$size"/apps/${pname}.png
-      fi
-    done;
+      icotool --extract $out/opt/STM32CubeMX/help/${iconame}.ico
+      fdupes -dN . > /dev/null
+      ls
+      for size in 16 24 32 48 64 128 256; do
+        mkdir -pv $out/share/icons/hicolor/"$size"x"$size"/apps
+        if [ $size -eq 256 ]; then
+          mv ${iconame}_*_"$size"x"$size"x32.png \
+            $out/share/icons/hicolor/"$size"x"$size"/apps/${pname}.png
+        else
+          convert -resize "$size"x"$size" ${iconame}_*_256x256x32.png \
+            $out/share/icons/hicolor/"$size"x"$size"/apps/${pname}.png
+        fi
+      done;
 
-    cp ${desktopItem}/share/applications/*.desktop $out/share/applications
-  '';
-
-  meta = with lib; {
-    description = "A graphical tool for configuring STM32 microcontrollers and microprocessors";
-    longDescription = ''
-      A graphical tool that allows a very easy configuration of STM32
-      microcontrollers and microprocessors, as well as the generation of the
-      corresponding initialization C code for the Arm® Cortex®-M core or a
-      partial Linux® Device Tree for Arm® Cortex®-A core), through a
-      step-by-step process.
+      cp ${desktopItem}/share/applications/*.desktop $out/share/applications
     '';
-    homepage = "https://www.st.com/en/development-tools/stm32cubemx.html";
-    sourceProvenance = with sourceTypes; [ binaryBytecode ];
-    license = licenses.unfree;
-    maintainers = with maintainers; [ angaz wucke13 ];
-    platforms = platforms.all;
+
+    meta = with lib; {
+      description = "A graphical tool for configuring STM32 microcontrollers and microprocessors";
+      longDescription = ''
+        A graphical tool that allows a very easy configuration of STM32
+        microcontrollers and microprocessors, as well as the generation of the
+        corresponding initialization C code for the Arm® Cortex®-M core or a
+        partial Linux® Device Tree for Arm® Cortex®-A core), through a
+        step-by-step process.
+      '';
+      homepage = "https://www.st.com/en/development-tools/stm32cubemx.html";
+      sourceProvenance = with sourceTypes; [ binaryBytecode ];
+      license = licenses.unfree;
+      maintainers = with maintainers; [ angaz wucke13 ];
+      platforms = platforms.all;
+    };
   };
+  in
+  buildFHSEnv {
+    inherit (package) pname meta;
+    runScript = "${package.outPath}/bin/stm32cubemx";
+    targetPkgs = pkgs:
+      with pkgs; [
+        alsa-lib
+        at-spi2-atk
+        cairo
+        cups
+        dbus
+        expat
+        glib
+        gtk3
+        libdrm
+        libGL
+        libudev0-shim
+        libxkbcommon
+        mesa
+        nspr
+        nss
+        pango
+        xorg.libX11
+        xorg.libxcb
+        xorg.libXcomposite
+        xorg.libXdamage
+        xorg.libXext
+        xorg.libXfixes
+        xorg.libXrandr
+      ];
 }

